tests/marine/benchmark/conversation_generators.py

```diff
@@ -0,0 +1,173 @@
+import os
+from collections import defaultdict
+from random import randint
+from typing import List, Callable, Dict, Set, Tuple
+
+from pypacker.layer12 import ethernet
+from pypacker.layer3 import ip
+from pypacker.layer4 import tcp, udp
+from pypacker.pypacker import Packet
+
+from .utils import Layer3Conversation, BenchmarkPacket
+
+
+class PortGenerator:
+    _conversation_to_ports: Dict[Layer3Conversation, set] = defaultdict(set)
+
+    @staticmethod
```

Comment:
You're still keeping as a sad global state.
Convert it to instance methods and make `generate_packets` initialize an instance and pass it to tcp/udp generators.
But is it really so crucial to generate distinct ports? Can't we just rely on randomness to minimize collisions and tolerate the rare few collisions that may occur?

Reply:
I'll use this comment to also comment on your general comment.

In terms of randomness in the benchmark - I think creating a "static case" will take just as much code (still need to generate the conversations - which is just as much code). We'll obviously have to keep this code if we want to extend the "static case", so in terms of total code usage we haven't achieved much.

Additionally, and in my opinion even more important, is treating Marine as a black box. We don't know how conversation storage and creation affects the benchmark, so using random values allows us to at least see we can get similar results over many runs, and indicate that the code works fine in terms of speed.

As for the ports, I think keeping the random is important. I agree with you that we can trust randomness to minimize collisions, so I'll remove the state.